### PR TITLE
Use the multislice view for single crystal diffraction

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This changes the technique-dependent initial view for a MDHistoworkspace from a single crystal diffraction instrument to the the MultiSlice view. This avoids the very slow `vtkDataSetSurfaceFilter` that is causing unacceptably slow start times.

I made additional changes to avoid switching to standard view for rendering.

**To test:**

1. In the MD Plotting preferences, change the initial view back to "technique-dependendent". Restart MantidPlot. 
2. Open a MDHistoworkspace from a SCD instrument such as TOPAZ, view it in the VSI and verify that it opens in the MultSlice view.
3. Verify there are no issues switching between views.
4. Open a MDEventWorkspace, view it in the VSI and verify that the SplatterPlot opens correctly and there are no issues switching views. 

There is no GitHub issue associated with this pull request.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by quantumsteve** [comment]
After switching from the multislice to the standard view, the camera position is off( zoomed in too far). Any idea what is causing this?

File: Vates/VatesSimpleGui/ViewWidgets/src/MdViewerWidget.cpp [hunk 1]
```diff
@@ -1243,10 +1232,7 @@ void MdViewerWidget::swapViews() {
   if (!this->hiddenView)
     g_log.error(
         "Inconsistency found when swapping views, the next view is NULL");
-
-  ViewBase *temp = this->currentView;
-  this->currentView = this->hiddenView;
-  this->hiddenView = temp;
+  std::swap(this->currentView, this->hiddenView);
```
Comment (AntonPiccardoSelg) [inline Vates/VatesSimpleGui/ViewWidgets/src/MdViewerWidget.cpp]: :+1:

**Comment by quantumsteve** [comment]
@VickieLynch Would you look at this again? I think the camera position issues are now resolved.

**Comment by VickieLynch** [comment]
Splatter plot is greyed out after multislice is drawn for me.  This worked this morning.

**Comment by quantumsteve** [comment]
@VickieLynch Is this with a MDEventWorkspace or MDHistoWorkspace?

File: Vates/VatesSimpleGui/ViewWidgets/src/MultisliceView.cpp [hunk 2]
```diff
@@ -100,6 +100,9 @@ void MultiSliceView::setupData() {
 
 void MultiSliceView::render() {
   this->origSrc = pqActiveObjects::instance().activeSource();
+  if (NULL == this->origSrc) {
```
Comment (AntonPiccardoSelg) [inline Vates/VatesSimpleGui/ViewWidgets/src/MultisliceView.cpp]: Could you use `nullptr`?

File: Vates/VatesSimpleGui/ViewWidgets/src/ThreesliceView.cpp [hunk 3]
```diff
@@ -85,6 +83,10 @@ void ThreeSliceView::makeThreeSlice() {
 
   this->origSrc = src;
 
+  if (NULL == this->origSrc) {
+    return;
```
Comment (AntonPiccardoSelg) [inline Vates/VatesSimpleGui/ViewWidgets/src/ThreesliceView.cpp]: Could you use `nullptr`?

File: Vates/VatesSimpleGui/ViewWidgets/src/MdViewerWidget.cpp [hunk 4]
```diff
@@ -871,14 +860,12 @@ MdViewerWidget::checkViewAgainstWorkspace(ModeControlWidget::Views view,
...
-          << "cannot be loaded into a splatterplot view. Defaulted to standard "
-             "view. \n";
-
-      selectedView = ModeControlWidget::STANDARD;
+      g_log.notice("The preferred initial view favours the splatterplot "
+                   "as initial view, but an MDHisto workspace is being "
+                   "loaded. A MDHisto workspace cannot be loaded into a "
+                   "splatterplot view. Defaulted to MultiSlice view.");
+
+      selectedView = ModeControlWidget::MULTISLICE;
```
Comment (AntonPiccardoSelg) [inline Vates/VatesSimpleGui/ViewWidgets/src/MdViewerWidget.cpp]: This change will affect all MDHisto workspaces where the user tries to load them initially into Splatterplot mode -- not only Single Crystal Diffraction. Is this intended?

**Comment by AntonPiccardoSelg** [comment]
Performed functional testing with SCD data sets and the behaviour is as expected. I am concerned however that the change will not only be affecting SCD.

**Comment by quantumsteve** [comment]
@AntonPiccardoSelg Yes, the change is a bit more general but I think that is OK. We want to avoid users being automatically switched to the relatively slow-to-render standard view.

**Comment by AntonPiccardoSelg** [comment]
@quantumsteve This might surprise some users. Although this should only affect users who have selected Splatterplot for MDHisto workspaces. I think that having it general probably makes sense. Do you think we should mention in the release notes that this is a general change and not a SCD-specific one?

**Comment by VickieLynch** [comment]
paraview
@quantumsteve See attached screenshot.  I cannot make scattterplot with this MDHistoWorkspace.

**Comment by AntonPiccardoSelg** [comment]
@VickieLynch That is by design. The SplatterPlot is disabled for MDHisto Workspaces. It should only be enabled if you are dealing with MDEvent Workspaces.

**Comment by VickieLynch** [comment]
@AntonPiccardoSelg You are right.  The old default was standard, not SplatterPlot.  This is working well for my large files.  :shipit:

**Pull request merged**